docs/LSPSupport.md

```diff
@@ -427,6 +427,12 @@ Here is an example with the [TypeScript Language Server](./user-defined-ls/types
 [LSPFormattingOnlyService](https://github.com/redhat-developer/lsp4ij/blob/main/src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPFormattingOnlyService.java) /
 [LSPFormattingAndRangeBothService](https://github.com/redhat-developer/lsp4ij/blob/main/src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPFormattingAndRangeBothService.java) with the `formattingService` extension point.
 
+### On-Type Formatting
+

```

Comment:
If a file is associated to a language server which have ontypeformating capability, is client on type formatting is disabled?

We need to clarify the behavoir somewhere.

Reply:
@angelozerr I'm not sure that I understand this question, but I'll try to answer as best I can. This capability is evaluated independent of other capabilities. If the language server claims to support the feature, this implementation will try to use it when one of the language server's reported trigger characters has been typed. Does that help?

Reply:
Sorry with my bad explanation.

You did an another PR which manage on type formatting on client side which uses selection ranges or folding.

Is this feature is disabled when the file is associated to a language server which supports ontypeformatting?

Reply:
Ah, that makes sense. No, right now these are two independent branches/PRs both based on `main`. Once the server-side PR is merged, I'll update the client-side PR accordingly. Instead of making them mutually-exclusive, though, I was thinking about making it so that the client-side typed handler is ordered before the server-side typed handler so that if the user preferred the behavior of client-side on-type formatting, it would supersede server-side on-type formatting if enabled. Obviously if client-side on-type formatting isn't enabled, it's a total no-op anyway.